### PR TITLE
WRN-18994: [Drawing POC] Undo/Redo Functionality

### PR DESCRIPTION
**Pull request by vladut-clapou-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Implemented undo/redo functionality for the Drawing component

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-18994

### CODE REVIEW

File: packages/ui/Drawing/Drawing.js [hunk 1]
```diff
@@ -239,46 +256,54 @@ const DrawingBase = kind({
...
 			setIsDrawing(false);
+			currentLinesArray.push(currentLine);
+
+			currentLine = {
+				points: [],
+				drawingTool: '',
+				fillColor: '',
+				brushColor: '',
+				brushSize: ''
+			};
```
Comment (daniel-stoian-lgp) [inline packages/ui/Drawing/Drawing.js:267]: here also

File: packages/ui/Drawing/utils.js [hunk 2]
```diff
@@ -66,6 +66,7 @@ const fillDrawing = (event, contextRef) => {
 			}
 		});
 	}
+
```
Comment (daniel-stoian-lgp) [inline packages/ui/Drawing/utils.js:69]: remove empty line

File: packages/ui/Drawing/utils.js [hunk 3]
```diff
@@ -84,7 +85,134 @@ const drawing = (beginPoint, controlPoint, endPoint, contextRef) => {
...
+		}
+		contextRef.current.closePath();
+		context.lineWidth = brushSize;
+		context.strokeStyle = brushColor;
+		context.fillStyle = fillColor;
+	}
+};
+
+
+
```
Comment (daniel-stoian-lgp) [inline packages/ui/Drawing/utils.js:210]: remove these 2 empty lines

**Review by daniel-stoian-lgp — COMMENTED** [review]
A few comments

File: packages/ui/Drawing/Drawing.js [hunk 4]
```diff
@@ -18,14 +18,22 @@ import {useImperativeHandle, useEffect, useRef, useState} from 'react';
 import ForwardRef from '../ForwardRef';
 import ri from '../resolution';
 
-import {drawing, fillDrawing} from './utils';
+import {drawing, drawCircle, drawRectangle, drawTriangle, fillDrawing, paint} from './utils';
```
Comment (stanca-pop-lgp) [inline packages/ui/Drawing/Drawing.js:21]: drawing, drawCircle - please sort

File: packages/ui/Drawing/Drawing.js [hunk 5]
```diff
@@ -239,46 +256,55 @@ const DrawingBase = kind({
...
 			contextRef.current.beginPath(); // start a canvas path
 			contextRef.current.moveTo(offsetX, offsetY); // move the starting point to initial position
 			points.push({x: offsetX, y: offsetY});
 
+			currentLine['points'].push({x: offsetX, y: offsetY});
+			currentLine['drawingTool'] = drawingTool;
+			currentLine['fillColor'] = fillColor;
+			currentLine['brushColor'] = brushColor;
+			currentLine['brushSize'] = brushSize;
+			currentLine['ev'] = ev;
```
Comment (stanca-pop-lgp) [inline packages/ui/Drawing/Drawing.js:291]: Could you please create a function that returns this so we don't repeat this code block?

File: packages/ui/Drawing/Drawing.js [hunk 6]
```diff
@@ -18,14 +18,22 @@ import {useImperativeHandle, useEffect, useRef, useState} from 'react';
...
 	pen: 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAATFJREFUeNrs19FthDAMBuB0A0bICIyQERiB2yAjsEFGyAiMABuEDa5MEDZwceWc0oq6J52dp7MU8YL4sCOiH2N0qzeNazoX0MqtXiAiGEKAZVlgGIaCj5poKGhd4ziW7p0G6unBue97yDlf4eJjj9RR4nAa+10M7boOYoyAVw7HPacXlOkUO8FKKbE47j2N+/VO8cGI4x5yOKFAn9prKCLYzTM4oVEELfUkHkRRDp+mqYzXq6BXuHMOVMbL4dbatigWfs9v9I3+efa2Ri2lhMcB0AI1VTz5F5dEbRXOWFwSxRoKxOHS6HdIw7P1VzD7gWugWIv3/ioVPnANFAvmeb4KZjUeJMEPipvpHKXZ992s62q2bTPHcZR71nPdzvWpkYehiqeROlT/7XBaCZ+rLwEGAPhaImYfzD7mAAAAAElFTkSuQmCC'
 };
+let currentLine = {
+		brushColor: '',
+		brushSize: '',
+		drawingTool: '',
+		fillColor: '',
+		ev: null,
+		points: []
+	}, currentLinesArray = [], actionsIndex = -1, lastAction = '';
```
Comment (stanca-pop-lgp) [inline packages/ui/Drawing/Drawing.js:36]: Maybe add these on separate lines:
currentLinesArray = [], 
actionsIndex = -1, 
lastAction = '';

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Pull request merged**